src/components/angular2-modal/models/dialog-ref.ts

```diff
@@ -1,20 +1,39 @@
 import { ComponentRef } from '@angular/core';
-import { PromiseWrapper } from '@angular/core/src/facade/promise';
+import { PromiseWrapper, PromiseCompleter } from '@angular/core/src/facade/promise';
 
 /**
  * API to an open modal window.
  */
 export class DialogRef<T> {
-    contentRef: ComponentRef<any>;
+    /** 
+     * The reference to the component ref.
+     * This can be return undefined, since the componentRef is set only when the modal is shown.
+     * Use componentRefPromise to get a promise that will resolve when the componentRet is set.
+     * @return {ComponentRef<any>}
+     */
+    get contentRef(): ComponentRef<any> { return this._contentRef; }
+    set contentRef(value: ComponentRef<any>) {
+        this._contentRef = value;
+        this._conponentRefDeferred.resolve(value);
+    }
+
+    /**
+     * A promise that is resolved when the component ref is set.
+     * @return {Promise<ComponentRef<any>>}
+     */
+    get contentRefPromise() { return this._conponentRefDeferred.promise; }
```

Comment:
The problem is this.
When you open a modal the open resolves before `contentRef` has been set 
```ts
modal.confirm().open().then(dialog => {
      console.log(dialog.contentRef); // This prints undefined
});
```

This adds a promise that fulfills when componentRef has been set.
I guess this could be solved the open promise by making it resolve only when componentRef has been set